code/game/objects/items/constructionpermits.dm

```diff
@@ -0,0 +1,55 @@
+/* construction permits. Think blueprints but accessible to all engies and does NOT count as the antag steal objective
+these cannot rename rooms that are in by default BUT can rename rooms that are created via blueprints/permit*/
+
+/obj/item/blueprints/construction_permit
+	name = "Construction Permit"
+	desc = "An electronic permit designed to register a room for the use of APC and air alarms"
+	icon = 'icons/obj/items.dmi'
+	icon_state = "permit"
+	attack_verb = list("attacks", "baps", "hits")
+	w_class = W_CLASS_TINY
+
+
+	can_rename_areas = list(AREA_BLUEPRINTS)
+
+/obj/item/blueprints/construction_permit/attack_self(mob/M as mob)
+	interact()
+	return
+
+/obj/item/blueprints/construction_permit/Topic(href, href_list)
+	if ((usr.restrained() || usr.stat || usr.get_active_hand() != src))
+		return 1
+	if(..())
+		return 1
+
+/obj/item/blueprints/construction_permit/interact()
+	var/area/A = get_area()
+	var/text = {"<HTML><head><title>[src]</title></head><BODY>
```

Comment:
1. `interact()` should take a `var/mob/user`
2. This mentions blueprints instead of construction permits everywhere
3. This was copy pasted from mommi prints wasn't it.